### PR TITLE
[build-utils] add framework slug and version to build output

### PR DESCRIPTION
**Pull request by lubakravche**
Adds framework to Lambda and edge build outputs so that we can distinguish which framework they originated from when certain features should be applied to specific frameworks.

Breaking up https://github.com/vercel/vercel/pull/9447 into two parts:
This PR introduces the frameworks type. Part 2: https://github.com/vercel/vercel/pull/9449

ticket: ED-131

x-ref: [slack channel](https://vercel.slack.com/archives/C042LHPJ1NX)

### CODE REVIEW

File: packages/build-utils/src/edge-function.ts [hunk 1]
```diff
@@ -44,6 +44,9 @@ export class EdgeFunction {
   /** Cronjob definition for the edge function */
   cron?: Cron;
 
+  /** The framework */
+  framework?: string;
```
Comment (ijjk) [inline packages/build-utils/src/edge-function.ts]: Should this be `{ slug: string, version?: string }` to match the others?

File: packages/build-utils/src/edge-function.ts [hunk 2]
```diff
@@ -44,6 +44,12 @@ export class EdgeFunction {
   /** Cronjob definition for the edge function */
   cron?: Cron;
 
+  /** The framework */
```
Comment (TooTallNate) [inline packages/build-utils/src/edge-function.ts:47]:
```suggestion
  /** The framework which created this function */
```

**Review by ijjk — DISMISSED** [review]
Thanks!

File: packages/build-utils/src/edge-function.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/build-utils/src/edge-function.ts:47]: Actually, can we stick this in `types.ts`? That way we can share the same interface (and doc comment) with both `EdgeFunction` and `Lambda`.

**Review by TooTallNate — CHANGES_REQUESTED** [review]

File: packages/build-utils/src/edge-function.ts (hunk 2 above)
Comment (lubakravche) [reply packages/build-utils/src/edge-function.ts:47]: `Framework` or `FunctionFramework`?

Comment (TooTallNate) [reply packages/build-utils/src/edge-function.ts:47]: Let's go with `FunctionFramework` since there's already a `Framework` type in `@vercel/frameworks`.

**Review by TooTallNate — APPROVED** [review]

**Comment by TooTallNate** [comment]
Thanks Luba!

**Pull request merged**